ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -84,22 +104,263 @@ where
     Ok(changed_keys)
 }
 
+fn act_on_transfers_to_eth<D, H>(
+    storage: &mut Storage<D, H>,
+    transfers: &[TransferToEthereum],
+) -> Result<BTreeSet<Key>>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    let mut changed_keys = BTreeSet::default();
+    // all keys of pending transfers
+    let prefix = BRIDGE_POOL_ADDRESS.to_db_key().into();
+    let mut pending_keys: HashSet<Key> = storage
+        .iter_prefix(&prefix)
+        .0
+        .map(|(k, _, _)| {
+            Key::from_str(k.as_str()).expect("Key should be parsable")
+        })
+        .filter(is_pending_transfer_key)
+        .collect();
+
+    // Remove the completed transfers from the bridge pool
+    for event in transfers {
+        let pending_transfer = event.into();
+        let key = get_pending_key(&pending_transfer);
+        if likely(storage.has_key(&key)?.0) {
+            _ = storage.delete(&key)?;
+            _ = pending_keys.remove(&key);
+        } else {
+            // The transfer should exist in the bridge pool
+            unreachable!()
```

Comment:
nit: you can add the message in the comment to the panic itself
```suggestion
            unreachable!("The transfer should exist in the bridge pool");
```